### ISSUE TITLE
How do i escape an @ char in a code example block?

### ISSUE THREAD

**Issue by DigitalBrainJS** [jsdoc2md/jsdoc-to-markdown#209]
I need to document a decorator with a usage example, but this breaks the parser because jsdoc2md trying to parse decorators in the example block as jsdoc tag.
    /**
     * type decorator
     * @param {BasicType} type
     * @returns {MethodDecorator}
     * @alias module:define-accessor2#type
     * @example
     * class Cat {
     *   @string
     *   name = '';
     *   @type(string|number)
     *   foo= 123;
     * }
     */
Since jsdoc2md render an example block as a markdown code block I can't use &commat; to avoid this problem.

**Comment by 75lb** [comment]
Does this comment block render correctly using [jsdoc](https://github.com/jsdoc/jsdoc)? Jsdoc2md sits downstream of jsdoc, so if it's broken in jsdoc it will be broken in jsdoc2md..

**Comment by DigitalBrainJS** [comment]
Yes, JSDoc has the same behavior (I thought the has its own parser because of jsoc-parser in the dependencies), but since it render docs as html it's possible to use html entries chars to avoid this issue. Unfortunately, since jsdoc2md renders examples as markdown code blocks without the ability to use html entries inside, I can't use this workaround.
So, is there really no way to do post-processing of comments data after jsdoc parser? Does this package fully rely on parsed jsdoc data and has no access to raw data to make some fix? For example, just render commat; html entry to @ before render it to a code block.

**Comment by 75lb** [comment]
I've looked into a workaround but unfortunately there's nothing I can do to stop jsdoc parsing the `@` character in example blocks. It's a [known issue](https://github.com/jsdoc/jsdoc/issues/1521). 

Generally, you can use `&commat;` anywhere in jsdoc comments except within a code block, where github's markdown parser will print it literally as `&commat;` instead of `@`.

**Comment by DigitalBrainJS** [comment]
But this package could just replace &commat; to @ and then output it to a markdown block of code. This would not be an ideal solution, but acceptable to have at least some workaround for this issue.

**Comment by woody34** [comment]
I was able to escape the @ sign by preceding it with a zero width joiner character... It's a bit of a hack. I'm not sure if it will cause issues down the road either.

char:

![image](https://user-images.githubusercontent.com/8930703/216667394-6bbea1a8-17b2-42c3-8110-dda56c135e91.png)

code:

```js
/**
 * @description handles the setup and teardown of connection for google cloud pubsub
 * @example
 *
 * ```
 * ‌@Connect() // TODO: remove zero width joiner char once jsdocs accommodates
 * connect(): ClientConfig {
 *   const projectId = this.configService.get('SOME_PROJECT_ID')
 *   const credentials = this.configService.get('SOME_CREDENTIALS')
 *   return { projectId, credentials }
 * }
 * ```
 * @returns void
 */
```

result:

![image](https://user-images.githubusercontent.com/8930703/216667168-fe0eb9ee-844e-4293-a882-513f6b0902c1.png)
